### PR TITLE
changed ok to ok:{ruleid}

### LINKED ISSUE #937: Convert remaining 'ok' tags to 'ok: <rule-id>'
Most of our rules in this repository have accompanying tests in the form of code files. For example, consider the following rule/test:

- Rule: https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/security/audit/insecure-transport/urllib/insecure-urlretrieve.yaml
- Test: https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/security/audit/insecure-transport/urllib/insecure-urlretrieve.py

The "test" portion of the file is the `ruleid` or `ok` comment. E.g. in the above Python test file, `ruleid: insecure-urlretrieve` says that Semgrep must find a line violating the `insecure-urlretrieve` rule on the next line. This protects against false negatives, i.e. instances where we expected a finding, but didn't have one. `ok: insecure-urlretrieve` says that Semgrep must _not_ find a line violating the `insecure-urlretrieve` rule on the next line. This protects against false positives, i.e. instances where we don't expect a finding, but have one.

The `ok: <rule-id>` functionality was recently added, so not all our tests are using it yet. In https://github.com/returntocorp/semgrep-rules/pull/933 I attempted to automate adding the tag where possible. However, I couldn't find an automated solution to fix all the tests, so we'll need some manual fixing. The following tests need manual fixing:

- `go/lang/correctness/overflow/overflow.go`
- `go/lang/correctness/useless-eqeq.go`
- `go/lang/security/audit/crypto/math_random.go`
- `go/lang/security/audit/net/dynamic-httptrace-clienttrace-ok.go`
- `go/lang/security/audit/net/pprof_good.go`
- `java/lang/security/audit/cookie-missing-httponly.java`
- `javascript/chrome-remote-interface/security/audit/chrome-remote-interface-injection.js`
- `javascript/express/security/express-vm-injection.js`
- `javascript/express/security/express-vm2-injection.js`
- `javascript/express/security/express-wkhtml-injection.js`
- `javascript/lang/security/audit/vm-injection.js`
- `javascript/playwright/security/audit/playwright-injection.js`
- `javascript/puppeteer/security/audit/puppeteer-injection.js`
- `python/cryptography/security/insecure-cipher-algorithms.py`
- `python/cryptography/security/insecure-hash-algorithms.py`
- `python/django/compatibility/django-2_0-compat.py`
- `python/django/correctness/string-field-null-checks.py`
- `python/flask/security/audit/secure-set-cookie.py`
- `python/lang/best-practice/pass-body.py`
- `python/lang/correctness/common-mistakes/is-comparison-string.py`
- `python/lang/maintainability/return.py`
- `python/lang/security/deserialization/pickle.py`
- `python/lang/security/insecure-hash-algorithms.py`
- `python/pycryptodome/security/insecure-cipher-algorithm.py`
- `python/pycryptodome/security/insecure-hash-algorithm.py`

Each of these files has multiple rules being tested simultaneously, which is why I couldn't find a great automated solution. Let's go through each of these and update each `ok` to the correct `ok: <rule-id>`.

Here's a rough game plan:

1. [Install Semgrep](https://github.com/returntocorp/semgrep#getting-started)
2. Clone `semgrep-rules`: `git clone git@github.com:returntocorp/semgrep-rules.git`
3. Run the tests before, e.g. `semgrep --test --test-ignore-todo semgrep-rules/go/lang/correctness/overflow/`
4. Fixup the test with `ok` -> `ok: <rule-id>`
5. Run the tests again after and note the additional true negative (`TN`) count
6. Repeat for all the test files
7. Open a PR against `semgrep-rules` with the improved tests

Other useful information:

- Semgrep `--test` functionality: https://github.com/returntocorp/semgrep/blob/develop/semgrep/semgrep/test.py
- Semgrep rule file format: https://github.com/returntocorp/semgrep/blob/develop/docs/configuration-files.md
- Semgrep pattern features: https://github.com/returntocorp/semgrep/blob/develop/docs/pattern-features.md

Let me know if you have any questions :+1:

### PR DESCRIPTION
**Pull request by AzureUmbra**
First PR with r2c! I hope I did this right!

### CODE REVIEW

**Review by DrewDennison — APPROVED** [review]
Nice!

**Comment by minusworld** [comment]
Nice. looks good 👍 

The conflicts are a byproduct of all the other changes happening right now... so you didn't do anything wrong there
I think we can fix the conflicts by:
1. Checking out the `develop` branch
1. `git pull`
1. Check out your working branch, `issue937`
1. `git rebase`

From here, it should (hopefully) resolve the conflicts. If not, someone can jump in and help clean up the conflicts.

**Comment by mschwager** [comment]
> Nice. looks good +1
> 
> The conflicts are a byproduct of all the other changes happening right now... so you didn't do anything wrong there
> I think we can fix the conflicts by:
> 
>     1. Checking out the `develop` branch
> 
>     2. `git pull`
> 
>     3. Check out your working branch, `issue937`
> 
>     4. `git rebase`
> 
> 
> From here, it should (hopefully) resolve the conflicts. If not, someone can jump in and help clean up the conflicts.

Things are a little hectic with all the different changes going in right now. I removed the conflicts :+1:

File: go/lang/correctness/overflow/overflow.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ func mainInt16Ex1() {
 }
 
 func mainInt16Ex2() {
-	// ok
+	// ok: interter-overflow-int16
```
Comment (underyx) [inline go/lang/correctness/overflow/overflow.go:19]: Huh, that should probably be spelled `integer` in the rule ID, right? Mind fixing that typo in the rule?

Comment (AzureUmbra) [reply go/lang/correctness/overflow/overflow.go:19]: Wilco, thanks!
```suggestion
	// ok: integer-overflow-int16
```

Comment (underyx) [reply go/lang/correctness/overflow/overflow.go:19]: Note it needs to be updated in the YAML as well to match! Probably worth creating a new PR for that.

**Pull request merged**